src/components/navigation.tsx

```diff
@@ -4,30 +4,31 @@ import { NavigationSectionData } from '../types';
 import { isSmallScreen } from '../util/isSmallScreen';
 
 type Props = {
-  sections: NavigationSectionData[];
+  sections: NavigationSectionData;
+  currentSlug: string;
 };
 
-const Navigation = ({ sections }: Props) => {
+const Navigation = ({ sections, currentSlug }: Props) => {
   const [isOpen, setIsOpen] = useState(false);
   const toggle = () => setIsOpen(!isOpen);
-  const onItemClick = () => isSmallScreen() && toggle();
+  const onItemClick = () => {
+    isSmallScreen() && toggle();
+  };
   const className = isOpen ? 'side-nav side-nav--open' : 'side-nav';
-
   return (
     <nav className={className}>
       <button className="side-nav__open" onClick={toggle}>
         Menu
       </button>
-      {sections.map((section: NavigationSectionData) => {
-        return (
-          <NavigationSection
-            title={section.title}
-            items={section.items}
-            key={section.title}
-            onItemClick={onItemClick}
-          />
-        );
-      })}
+      {Object.keys(sections).map(section => (
+        <NavigationSection
+          key={section}
+          title={section}
+          section={sections[section]}
```

Comment:
VSCode tells me that: Type 'NavigationSectionItem[][]' is not assignable to type 'NavigationSectionItem[]. Maybe some minor tweaking of the types are needed?

Reply:
Mh. I can't make a good fist of that one. Any suggestions?

Reply:
Should be fixed by doing,
```typescript
export interface NavigationSectionData {
  [index: string]: NavigationItemList;
}
```